coverage_comment/storage.py

```diff
@@ -54,52 +35,50 @@ def checked_out_branch(git: subprocess.Git, branch: str):
 
     git.fetch()
 
-    branch_exists = True
+    log.debug("Resetting all changes")
+    # Goodbye `.coverage` file.
+    git.reset("--hard")
+
     try:
-        git.checkout(branch)
-        log.debug(f"Branch {branch} exist.")
+        git.rev_parse("--verify", branch)
```

Comment:
The problem seems to be that the action does not recognize the remote branch and therefore creates its own. This is because rev-parse does not seem to work for a branch that does not also exist locally. We need to either checkout again to find out if the branch exists remotely or find another method.

Reply:
I'm going to test, but I think verifying `origin/{branch}` instead of `branch` might suffice.